gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/UserQuotaManager.java

```diff
@@ -194,4 +217,12 @@ private int getQuotaForUser(String user) {
     return this.perUserQuota.getOrDefault(user, defaultQuota);
   }
 
+  private int getQuotaForFlowGroup(String flowGroup) {
+    return this.perFlowGroupQuota.getOrDefault(flowGroup, defaultQuota);
+  }
+
+  public boolean isDagCurrentlyRunning(Dag.DagNode<JobExecutionPlan> dagNode) {
```

Comment:
should we use this function to decrement/increment quotas?

Reply:
Oh good catch, I forgot to use this function in a retry scenario. I can add a test covering that case too.